lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -430,6 +438,9 @@ def _fill_nodes_metadata(
                 private_key_file=self.runbook.admin_private_key_file,
             )
 
+            # Ensure cloud-init completes its setup.
+            node.execute("cloud-init status --wait", sudo=True, expected_exit_code=0)
```

Comment:
for each `expected_exit_code`, it's better to specify message too. It helps on troubleshooting.